src/test/java/bio/terra/service/snapshot/flight/SnapshotAuthzIamStepTest.java

```diff
@@ -104,4 +110,73 @@ void testDoAndUndoStepWithDUOS() throws InterruptedException {
     assertThat(undoResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_SUCCESS));
     verify(iamService).deleteSnapshotResource(TEST_USER, SNAPSHOT_ID);
   }
+
+  @Test
+  void testDoAndUndoWithSnapshotFirecloudGroup() throws InterruptedException {
+    workingMap.put(
+        SnapshotWorkingMapKeys.SNAPSHOT_FIRECLOUD_GROUP_EMAIL, SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+    snapshotRequestModel.contents(
+        List.of(
+            new SnapshotRequestContentsModel()
+                .mode(SnapshotRequestContentsModel.ModeEnum.BYREQUESTID)));
+    step =
+        new SnapshotAuthzIamStep(
+            iamService, snapshotService, snapshotRequestModel, TEST_USER, SNAPSHOT_ID);
+    StepResult doResult = step.doStep(flightContext);
+    assertThat(doResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_SUCCESS));
+
+    ArgumentCaptor<SnapshotRequestModelPolicies> argument =
+        ArgumentCaptor.forClass(SnapshotRequestModelPolicies.class);
+    verify(iamService).createSnapshotResource(eq(TEST_USER), eq(SNAPSHOT_ID), argument.capture());
+    List<String> readers = argument.getValue().getReaders();
+    assertTrue(readers.contains(SNAPSHOT_FIRECLOUD_GROUP_EMAIL));
```

Comment:
I don't think a captor is needed here, since we know all the arguments passed to `createSnapshotResource()`
```suggestion
    var policies = new SnapshotRequestModelPolicies().addReadersItem(SNAPSHOT_FIRECLOUD_GROUP_EMAIL);
    verify(iamService).createSnapshotResource(TEST_USER, SNAPSHOT_ID, policies);
```
Also, I see that none of these tests are asserting that the step correctly populates `SnapshotWorkingMapKeys.POLICY_MAP`. If you wanted to add that here, you could replace the `verify()` with a `when()` and then assert that the when-generated result is stored in the flight map.

Reply:
Ok - I've reworked this to test the policy_map and to change from `verify` to `when`.